Dockerfile

```diff
@@ -1,10 +1,10 @@
-FROM python:3.6.0-slim
+FROM python:3.6-slim@sha256:5a96684a1729acd0680b39a3c24ef33f36bb6951873f77cde1e227b059a0f881
 WORKDIR /app
 RUN groupadd --gid 1001 app && useradd -g app --uid 1001 --shell /usr/sbin/nologin app
 RUN apt-get update && \
     apt-get upgrade -y && \
     apt-get install -y --no-install-recommends \
-    gcc libpq-dev curl apt-transport-https libffi-dev openssh-client
+    gcc libpq-dev curl apt-transport-https libffi-dev openssh-client gnupg
 
 # Install node from NodeSource
```

Comment:
Are the steps below really the "best" way to do this? It doesn't look much like the various Dockerfiles I found for [node 8 for stretch](https://github.com/nodejs/docker-node/blob/master/8/stretch/Dockerfile).

Reply:
For example, when you build the image you get warning messages like this:
```
Warning: apt-key output should not be parsed (stdout is not a terminal)
```

Reply:
It makes sense, but it's still sadface to have to look at warnings.

Reply:
We aren't parsing the contents of apt-key. That error happens any time that the output doesn't go to an interactive terminal, and our CI setup is not interactive.

I'm sure these steps aren't the best way to do this. The best way would be to maintain a combination Python-and-Node image and pull that instead of hacking up the Python image to add Node. But it works, so I'm not inclined to change it. If you'd like to do so, go ahead.